src/ovp_pipeline/commands/refresh_source_authority.py

```
@@ -0,0 +1,287 @@
+        fd = os.open(lock_path, os.O_CREAT | os.O_EXCL | os.O_WRONLY)
+    except FileExistsError:
+        # Steal stale lock if its PID is dead.
+        try:
+            existing_pid = int(lock_path.read_text(encoding="utf-8").strip())
+        except (OSError, ValueError):
+            existing_pid = -1
+        if existing_pid > 0 and _pid_alive(existing_pid):
+            raise SystemExit(
+                f"another refresh is running (pid {existing_pid}); "
+                f"remove {lock_path} after confirming it crashed",
+            )
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)